src/libs/ReportUtils.js

```diff
@@ -1847,6 +1847,28 @@ function getWhisperDisplayNames(participants) {
     return _.map(participants, (login) => getDisplayNameForParticipant(login, !isWhisperOnlyVisibleToCurrentUSer)).join(', ');
 }
 
+function buildOptimisticCancelTaskReportAction(taskReportID) {
+    return {
+        taskReportID,
+        actionName: CONST.REPORT.ACTIONS.TYPE.TASK.CANCEL,
+        actorAccountID: currentUserAccountID,
+        actorEmail: currentUserEmail,
+        automatic: false,
+        avatar: lodashGet(currentUserPersonalDetails, 'avatar', getDefaultAvatar(currentUserEmail)),
+        isAttachment: false,
+        person: [
+            {
+                style: 'strong',
+                text: lodashGet(currentUserPersonalDetails, 'displayName', currentUserEmail),
+                type: 'TEXT',
+            },
+        ],
+        reportActionID: NumberUtils.rand64(),
+        shouldShow: true,
+        created: DateUtils.getDBTime(),
+    };
```

Comment:
@thienlnam how does this optimistic data look? No pending action needed for this right?

Reply:
nice nice, looks solid!

Reply:
Actually, should this cancel action just be an ADDCOMMENT like it is when a Task is created [here](https://github.com/infinitered/ExpensifyApp/blob/5df596f1be385c9767aef239fd24a6d2fd7771e7/src/libs/actions/Task.js#L46)? Not entirely sure how to go about creating and rendering this action

Reply:
Not quite, it should still be a separate reportAction. It would look similar to the RenameAction